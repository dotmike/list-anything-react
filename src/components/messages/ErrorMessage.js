import React from "react";
import PropTypes from "prop-types";
import { Message } from "semantic-ui-react";

const ErrorMessage = ({ text, showHeader }) => (
  <Message negative style={{ textAlign: "center" }}>
    {showHeader && <Message.Header>Something went wrong</Message.Header>}
    <p>{text}</p>
  </Message>
);

ErrorMessage.propTypes = {
  // ownProps
  text: PropTypes.string.isRequired,
  showHeader: PropTypes.bool
};

ErrorMessage.defaultProps = {
  showHeader: true
};

export default ErrorMessage;
