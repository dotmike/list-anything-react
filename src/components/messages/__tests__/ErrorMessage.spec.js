import React from "react";
import { shallow } from "enzyme";

import ErrorMessage from "../ErrorMessage";

describe("ErrorMessage", () => {
  it("renders correctly when `showHeader` is true", () => {
    const wrapper = shallow(<ErrorMessage text="some text" showHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when `showHeader` is false", () => {
    const wrapper = shallow(
      <ErrorMessage text="some text" showHeader={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
