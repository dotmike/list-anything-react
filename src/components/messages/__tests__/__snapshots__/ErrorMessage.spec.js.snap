// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly when \`showHeader\` is false 1`] = `
<Message
  negative={true}
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <p>
    some text
  </p>
</Message>
`;

exports[`ErrorMessage renders correctly when \`showHeader\` is true 1`] = `
<Message
  negative={true}
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <MessageHeader>
    Something went wrong
  </MessageHeader>
  <p>
    some text
  </p>
</Message>
`;
